### PR TITLE
Adding Link SemanticsFlag

### PR DESCRIPTION
**Pull request by Piinks**
Framework counter PR: https://github.com/flutter/flutter/pull/41327 - will require a manual engine roll.

Part of https://github.com/flutter/flutter/issues/36306

### CODE REVIEW

File: shell/platform/embedder/embedder.h [hunk 1]
```diff
@@ -170,6 +169,8 @@ typedef enum {
   ///
   /// Only applicable when kFlutterSemanticsFlagIsTextField flag is on.
   kFlutterSemanticsFlagIsReadOnly = 1 << 20,
+  /// Whether the semantics node represents a link.
+  kFlutterSemanticsFlagIsLink = 1 << 21,
```
Comment (Piinks) [inline shell/platform/embedder/embedder.h]: Should this be 22 like the rest? I noticed `isFocusable` is not represented here.
cc/ @gspencergoog

Comment (gspencergoog) [reply shell/platform/embedder/embedder.h]: Yes, I that was an oversight, thanks!  Created https://github.com/flutter/engine/pull/12700 to fix that.

File: lib/ui/semantics.dart [hunk 2]
```diff
@@ -349,6 +350,14 @@ class SemanticsFlag {
   /// Only applicable when [isTextField] is true.
   static const SemanticsFlag isReadOnly = SemanticsFlag._(_kIsReadOnlyIndex);
 
+  /// Whether the semantic node is an interactive link.
+  ///
+  /// Platforms have special handling for links, for example Android's TalkBack
```
Comment (goderbauer) [inline lib/ui/semantics.dart]: lets only talk about iOS here since Android doesn't really have this.

**Comment by Piinks** [comment]
Landing this on red to do a manual engine roll for framework PR https://github.com/flutter/flutter/pull/41327

**Comment by Piinks** [comment]
I'm going to revert this since I cannot complete a manual engine roll with the current status of the framework build. I will try again next week.

**Pull request merged**